lib/workers/branch/index.ts

```diff
@@ -320,6 +326,60 @@ export async function processBranch(
     } else {
       logger.debug('No updated lock files in branch');
     }
+
+    if (global.trustLevel === 'high') {
+      logger.debug('Executing post-change hooks');
+      if (
+        is.nonEmptyArray(config.postUpgradeTasks) &&
+        is.nonEmptyArray(global.allowedPostUpgradeTasks)
+      ) {
+        for (const cmd of config.postUpgradeTasks) {
+          if (
+            !_.some(global.allowedPostUpgradeTasks, (pattern: string) =>
+              cmd.match(pattern)
+            )
+          ) {
+            logger.debug(
+              { cmd, allowedPostUpgradeTasks: global.allowedPostUpgradeTasks },
+              'Post-upgrade task did not match any on allowed list'
+            );
+          } else {
+            logger.debug({ cmd }, 'Executing post-change hook');
+
+            const execResult = await exec(cmd, {
+              cwd: config.localDir,
+            });
+
+            logger.debug(execResult, 'Executed post-change hook');
+          }
+        }
+      }
+
+      if (is.nonEmptyArray(config.postUpgradeFiles)) {
```

Comment:
I had expected this logic would only be used if a postUpgradeTask was executed. Is it kept separate for any real use case that exists today?

Reply:
Yes, that was the intention. I've shifted this block so it follows the post-upgrade task execution rather than being in a separate block